calcite-adapter/src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java

```diff
@@ -577,24 +599,12 @@ private static Operand getMongoAggregateForOperator(
             final RexCall call,
             final List<Operand> strings,
             final String stdOperator) {
-        verifySupportedType(call);
         if (hasObjectIdAndLiteral(call, strings)) {
             return new Operand(getObjectIdAggregateForOperator(call, strings, stdOperator));
         }
         return new Operand("{" + maybeQuote(stdOperator) + ": [" + Util.commaList(strings) + "]}");
     }
 
-    private static void verifySupportedType(final RexCall call)
```

Comment:
moved this check to be under trandlateDateAdd specifically since we do support simple interval + interval addition for months and years.

Reply:
All good.